doctr/models/builder.py

```diff
@@ -45,34 +45,37 @@ def _sort_boxes(boxes: np.ndarray) -> np.ndarray:
         """Sort bounding boxes from top to bottom, left to right
 
         Args:
-            boxes: bounding boxes of shape (N, 4) or (N, 5) (in case of rotated bbox)
+            boxes: bounding boxes of shape (N, 4) or (N, 4, 2) (in case of rotated bbox)
 
         Returns:
             indices of ordered boxes of shape (N,)
         """
-        if boxes.shape[1] == 5:
+        if len(boxes.shape) == 3: 
             boxes = rotate_boxes(
-                loc_preds=np.concatenate((boxes, np.zeros((boxes.shape[0], 1))), -1),
-                angle=-np.median(boxes[:, 4]),
+                loc_preds=boxes,
+                angle=-estimate_page_angle(boxes),
                 orig_shape=(1024, 1024),
                 min_angle=5.,
             )
-            return (boxes[:, 0] + 2 * boxes[:, 1] / np.median(boxes[:, 3])).argsort()
+            # Points are in this order: top left, top right, bot right, bot left
+            return (boxes[:, 0, 0] + 2 * boxes[:, 2, 1] / np.median(boxes[:, 2, 1] - boxes[:, 1, 1])).argsort()
```

Comment:
So there are a few things at hand here!

First, I believe you meant to write:
```python
np.median(boxes[:, 1, 1] - boxes[:, 0, 1])
```
because we're on (N, 4, 2), not (N, 2, 4)

Then, I believe there is an issue in the case of rotated ones:
- with the previous one, the `np.median(boxes[:, 3])` was yielding the median height
- but now we take `np.median(boxes[:, 2, 1] - boxes[:, 1, 1])` which is the median Y-diff between the supposedly top left and bottom left corners. But if the box is rotated, that value might not even be positive, and if it's rotated, we need the 2D distance between the two points

So I think it's actually:
```python
np.median(np.linalg.norm(boxes[:, 1] - boxes[:, 0], axis=-1))
```

What do you think? (for the 2nd point, I'm not sure whether the boxes have been straightened yet, but my suggestion should work in either case)